### PR TITLE
ocamlPackages.{h2, hpack}: init at 0.8.0

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

Dependencies of `git-paf` 3.4.0.

Mostly based on <https://github.com/anmonteiro/nix-overlays/blob/master/ocaml/h2/default.nix>. I want to add the other h2-* packages as well, but not motivated currently since this also entails packaging all gluten-* packages.

@anmonteiro I'd be happy to add you as maintainer as well if you want to.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
@ofborg build ocamlPackages.h2 ocaml-ng.ocamlPackages_4_12.h2

File: pkgs/development/ocaml-modules/h2/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  propagatedBuildInputs = [
+    angstrom
+    faraday
+    base64
+    psq
+    hpack
+    httpaf
+  ];
+
+  doCheck = true;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/h2/default.nix:43]: Build fail with OCaml 4.06.

File: pkgs/development/ocaml-modules/h2/default.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+, base64
+, psq
+, httpaf
+, alcotest
+, yojson
+, hex
+}:
+
+let
+  http2-frame-test-case = fetchFromGitHub {
```
Comment (anmonteiro) [inline pkgs/development/ocaml-modules/h2/default.nix:15]: why prefer this instead of pulling the submodules?

**Comment by anmonteiro** [comment]
@sternenseemann thanks for doing this one! Happy to be included in the maintainers list.

File: pkgs/development/ocaml-modules/h2/default.nix (hunk 2 above)
Comment (sternenseemann) [reply pkgs/development/ocaml-modules/h2/default.nix:15]: I wanted to use the release tar ball if possible which mainly motivated this. If you don't actually rely on `dune subst` or the likes we can probably use `fetchSubmodules = true`.

File: pkgs/development/ocaml-modules/h2/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/ocaml-modules/h2/default.nix:43]: Done!

**Pull request merged**